File: Contests/2019-2020/Bronze/Jan/race.cpp
#include <bits/stdc++.h>
#pragma GCC optimize(2)
using namespace std;
typedef long long ll;
const int MAXN = 1005;

int k, n, t;

int solve(int lim) {
    int l = 0, r = 0, time = 0, dist = 0;
    for (int cur = 1; ; ++cur) {
        dist += cur;
        time++;
        if (dist >= k) return time;
        if (cur >= lim) {
            dist += cur;
            time++;
            if (dist >= k) return time;
        } 
    }
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    freopen("race.in", "r", stdin);
    freopen("race.out", "w", stdout);
    
    cin >> k >> n;
    while (n--) {
        cin >> t;
        cout << solve(t) << endl;
    }
    return 0;
}

File: Contests/2019-2020/Bronze/Jan/race.py
import sys
sys.stdin = open("race.in", "r")
sys.stdout = open("race.out", "w")

k, n = map(int, input().split())

def solve(lim):
    time, dist = 0, 0
    cur = 1
    for cur in range(1, k):
        dist += cur
        time += 1
        if dist >= k:
            return time
        if cur >= lim:
            dist += cur
            time += 1
            if dist >= k:
                return time

for i in range(n):
    print(solve(int(input())))